webapi/src/main/scala/org/knora/webapi/store/triplestore/impl/TriplestoreServiceLive.scala

```diff
@@ -143,23 +154,7 @@ case class TriplestoreServiceLive(
     def parseJsonResponse(sparql: String, resultStr: String): IO[TriplestoreException, SparqlSelectResult] =
       ZIO
         .attemptBlocking(resultStr.parseJson.convertTo[SparqlSelectResult])
-        .foldZIO(
-          _ =>
-            if (resultStr.contains("##  Query cancelled due to timeout during execution")) {
-              ZIO.logError("Triplestore timed out while sending a response, after sending statuscode 200.") *>
-                ZIO.fail(
-                  TriplestoreTimeoutException(
-                    "Triplestore timed out while sending a response, after sending statuscode 200."
-                  )
-                )
-            } else {
-              ZIO.logError(
-                s"Couldn't parse response from triplestore:$logDelimiter$resultStr${logDelimiter}in response to SPARQL query:$logDelimiter$sparql"
-              ) *>
-                ZIO.fail(TriplestoreResponseException("Couldn't parse Turtle from triplestore"))
-            },
-          ZIO.succeed(_)
-        )
+        .foldZIO(_ => checkResponse(sparql, resultStr), ZIO.succeed(_))
```

Comment:
could this be done with `mapError`, so that you don't have to do `ZIO.succeed(_)`?

Reply:
I agree with Balduin here.
Please also consider to rename `checkResponse` method. 
Essentially it's primary use is turning the response into an exception`Triplestore*Exception`. The `response` does not really he to be check, as we know have an error already.

Reply:
It didn't work with `mapError` but `orEsle`did the job. I also renamed that method. Thanks!